src/store/tickets/thunk.ts

```diff
@@ -0,0 +1,66 @@
+import { Notify } from '../../utils/notify';
+import TicketService from '../../service/TicketService';
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
+import { ITickets, IUpdateTickets } from '../../types/tickets';
+
+export const fetchTicketAsync = createAsyncThunk('tickets/fetchTicket', async (apiUrl: string) => {
+  try {
+    const response = await TicketService.getAll(apiUrl);
+    const data = response.data;
+    const totalCount = parseInt(response.headers['x-total-count']);
+
+    return { data, totalCount };
+  } catch (error) {
+    Notify('Ticket creation error');
+  }
+});
+export const createTicketAsync = createAsyncThunk(
+  'tickets/createTicket',
+  async ({ apiUrl, data }: { apiUrl: string; data: ITickets }, thunkAPI) => {
+    const { dispatch } = thunkAPI;
+    const transformedData = {
+      ...data,
+      createdBy: {
+        name: 'userName',
```

Comment:
Please leave a comment to replace it with dynamic value when we enable auth flow.